src/library/scala/annotation/unused.scala

```diff
@@ -0,0 +1,16 @@
+/*
+ * Scala (https://www.scala-lang.org)
+ *
+ * Copyright EPFL and Lightbend, Inc.
+ *
+ * Licensed under Apache License 2.0
+ * (http://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package scala.annotation
+
+/** An annotation that designates that the target is unused. */
```

Comment:
The doc should say
  - why this extends `deprecated` and is not unrelated
  - what the compiler does with declarations annotated `@unused`
  - what it is recommended to be used for